src/test/encoding/readable.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python3
+import json
+import os
+import subprocess
+import tempfile
+import difflib
+from pathlib import Path
+import concurrent.futures
+from functools import partial
+from collections import OrderedDict
+
+CEPH_ROOT = os.getenv("CEPH_ROOT", "..")
+CEPH_DENCODER = os.path.join(CEPH_ROOT, "build/bin/ceph-dencoder")
+DIR = Path(os.path.join(CEPH_ROOT, "ceph-object-corpus"))
+
+temp_unrec = tempfile.mktemp(prefix=f"unrecognized_")
+err_file_rc = tempfile.mktemp(prefix=f"dencoder_err_")
+
+current_ver = subprocess.check_output([CEPH_DENCODER, "version"]).decode().strip()
+fast_shouldnt_skip = []
+backward_compat = {}
+incompat_paths = {}
+
+def sort_values(obj):
+    if isinstance(obj, dict):
+        return OrderedDict((k, sort_values(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return sorted(obj, key=sort_list_values)
+    return obj
+
+def sort_list_values(obj):
+    if isinstance(obj, dict):
+        return sorted(obj.items())
+    if isinstance(obj, list):
+        return sorted(obj, key=sort_list_values)
+    return obj
+
+
+def process_type(file_path, type):
+    print(f"dencoder test for {file_path}")
+    cmd1 = [CEPH_DENCODER, "type", type, "import", file_path, "decode", "dump_json"]
+    cmd2 = [CEPH_DENCODER, "type", type, "import", file_path, "decode", "encode", "decode", "dump_json"]
```

Comment:
I think these 2 lines are the crux of the entire idea, and thus deserve more intention-conveying names (or a comment). Something like `cmd_decode_recorded_bytestream` and `cmd_decode_reencoded_bytestream`? Still, they are not so good.